core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1300,6 +1300,8 @@ private boolean rerunWF(String workflowId, String taskId, Map<String, Object> ta
                     executionDAOFacade.removeTask(task.getTaskId());
                 }
             }
+            //reset start time of rerunTask
+            rerunFromTask.setStartTime(System.currentTimeMillis());
```

Comment:
Here, the task is only set to `SCHEDULED` state and not started hence startTime must be 0. It is only set to `IN_PROGRESS` and started if it is a SUB_WORKFLOW.
This should be added at line 1307.

Since rerunning a task basically means that the task is being restarted, other fields within the task also need to be reset.
```java
rerunFromTask.setScheduledTime(System.currentTimeMillis());
rerunFromTask.setStartTime(0);
rerunFromTask.setUpdateTime(0);
rerunFromTask.setEndTime(0);
rerunFromTask.setOutputData(null);
```

Reply:
Just reiterating, we need to reset all the fields to default but reset starttime to currenttime only if this task is a SUB_WORKFLOW.

Reply:
Also, need to reset `rerunFromTask.setExternalOutputPayloadStoragePath(null);`